R/calc_AUC.R

```diff
@@ -0,0 +1,99 @@
+require(caTools)
+
+calc_AUC <-
+  function(qp,
+           test_matrix_Xi,
+           test_matrix_Xip,
+           test_yi) {
+    rankdiff <- function(qp, matrix_Xi, matrix_Xip, t) {
+      Xirank <- qp$rank(X = matrix_Xi)
+      Xiprank <- qp$rank(X = matrix_Xip)
+      Xirank <- cbind(Xirank, Xiprank)
+      
+      Xirank <- data.frame(Xirank)
+      Xirank$diff <- 0
+      
+      for (i in 1:nrow(Xirank)) {
+        diff <- Xirank[i, 2] - Xirank[i, 1]
+        if (diff < -1 * t) {
+          Xirank[i, 3] <- -1
+        }
+        if (diff > t) {
+          Xirank[i, 3] <- 1
+        }
+        if (abs(diff) <= t) {
+          Xirank[i, 3] <- 0
+        }
+      }
+      return(Xirank)
+    }
+    
+    out_AUC <- data.frame(tau = 0,
+                          TP = 0,
+                          FP = 0,
+                          i = 0)
+    
+    rankdiff_out <-
+      rankdiff(qp, test_matrix_Xi, test_matrix_Xip,t = 1)
+    rankdiff_out$difference <- abs(rankdiff_out$X2 - rankdiff_out$X1)
+    
+    tau_array <- c(abs(rankdiff_out$difference))
+    
+    out_AUC_SVM <- data.frame(tau = 0, TP = 0, FP = 0)
```

Comment:
it would be faster to use a list of data frames, rather than rbind(small, big) at each iteration of the for loop, see https://github.com/tdhock/idioms#accumulating-dataframes